addons/police/functions/fnc_spawnVehicle.sqf

```diff
@@ -27,13 +29,30 @@ private _spawnPoints = if (_vehicleType isEqualTo "Helicopter" || {_vehicleType
     +(_spawner getVariable QGVAR(spawnPoints))
 };
 
-// Find empty spawn position
+// Find spawn position
 private _position = [];
 private _direction = 0;
-while {_position isEqualTo [] && {!(_spawnPoints isEqualTo [])}} do {
-    private _spawnPoint = [_spawnPoints] call EFUNC(common,deleteAtRandom);
-    private _objects = (getPos _spawnPoint) nearEntities 5;
-    if (_objects isEqualTo []) exitWith {
+
+private _emptySpawnPointIndex = _spawnPoints findIf {getPos _x nearEntities SPAWNPOINT_SAFEZONE isEqualTo []};
+if (_emptySpawnPointIndex isNotEqualTo -1) then {
+    private _spawnPoint = _spawnPoints select _emptySpawnPointIndex;
+    _position = getPos _spawnPoint;
+    _direction = getDir _spawnPoint;
+} else {
+    // Maybe there is a position that has unoccupied vehicle
+    private _fullSpawnPointsWithoutCrew = _spawnPoints select {
+        private _nearEntities = getPos _x nearEntities SPAWNPOINT_SAFEZONE;
+        if (_nearEntities isEqualTo []) exitWith { false };
+        _nearEntities findIf {crew _x isEqualTo []} isEqualTo -1
+    };
+
+    if (_fullSpawnPointsWithoutCrew isNotEqualTo []) exitWith {
+        private _spawnPoint = [_fullSpawnPointsWithoutCrew] call EFUNC(common,deleteAtRandom);
+
+        // Clear the area
+        _spawnPoint nearEntities SPAWNPOINT_SAFEZONE
```

Comment:
I would suggest using something like `sizeOf _vehicleClassname + SPAWNPOINT_SAFEZONE`

Reply:
This might work only sometimes.
> At least one object of the given classname has to be present in the current mission otherwise zero will be returned.
> https://community.bistudio.com/wiki/sizeOf
So it seems like I'll still need some const anyway as probably most vehicles available for spawn will not be on the map until spawned. Not sure if adding such handling is worth it. Although I'll see what I can do.

Reply:
I'll leave this as is for now, I've creted #67, so it can be handled later.